app/models/doc_categories/sidebar_link.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module DocCategories
+  class SidebarLink < ActiveRecord::Base
+    self.table_name = "doc_categories_sidebar_links"
+
+    belongs_to :sidebar_section,
+               class_name: "DocCategories::SidebarSection",
+               foreign_key: :doc_categories_sidebar_section_id,
+               inverse_of: :sidebar_links
+
+    belongs_to :topic, optional: true
+
+    validates :doc_categories_sidebar_section_id, presence: true
+    validates :position, presence: true, uniqueness: { scope: :doc_categories_sidebar_section_id }
+    validates :title, length: { maximum: 255 }
+    validates :href, length: { maximum: 2000 }
+    validate :target_present
+    validate :topic_not_deleted
+
+    private
+
+    def target_present
+      return if topic_id.present? || href.present?
```

Comment:
Should we ensure `topic_id` and `href` are mutually exclusive? This condition allows both to be set.